File: agent-service/app/api/v1/health.py
from datetime import datetime, timezone
from fastapi import APIRouter
from pydantic import BaseModel
from typing import Dict, Any

from app.core.config import settings

router = APIRouter()


class HealthResponse(BaseModel):
    status: str
    service: str
    version: str
    timestamp: str
    details: Dict[str, Any]


@router.get("/health", response_model=HealthResponse)
async def get_health():
    return HealthResponse(
        status="UP",
        service=settings.PROJECT_NAME,
        version=settings.VERSION,
        timestamp=datetime.now(timezone.utc).isoformat(),
        details={
            "environment": settings.ENVIRONMENT,
            "llm_provider": settings.LLM_PROVIDER,
            "intelligence_plane": "active",
        },
    )

File: frontend/src/types/health.ts
export interface ServiceHealth {
  status: 'UP' | 'DOWN' | 'PENDING';
  service: string;
  version?: string;
  timestamp?: string;
  details?: Record<string, any>;
  latencyMs?: number;
  error?: string;
}

export interface PlatformHealthState {
  controlPlane: ServiceHealth;
  intelligencePlane: ServiceHealth;
  lastChecked: string;
}
